### PR TITLE
Revamp android input handling to support mouse, pen and historical points

### PR DESCRIPTION
**Pull request by maxkatz6**
## What is the current behavior?

Only touch input is handled. Mouse/pen pointer types are not complete or not handled properly.

## What is the updated/expected behavior with this PR?

Touch, mouse and pen are handled with some additional parameters (pressure, meta keys...).
Historical/intermediate points are available too.
Also fixed inertia scrolling on android backend.

## Place to improve in the future

Possibly we should handle android input using NDK instead of Java APIs.
Mouse capture API is available but can't be tested on my phone, so I didn't push that.

### CODE REVIEW

**Comment by maxkatz6** [comment]
https://user-images.githubusercontent.com/3163374/183359403-45fa3146-9a03-4f3e-a68c-1410db7f782d.mp4

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0022994-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

File: src/Avalonia.Base/Input/GestureRecognizers/ScrollGestureRecognizer.cs [hunk 1]
```diff
@@ -66,8 +66,7 @@ public void Initialize(IInputElement target, IGestureRecognizerActionsDispatcher
         
         public void PointerPressed(PointerPressedEventArgs e)
         {
-            if (e.Pointer.IsPrimary && 
-                (e.Pointer.Type == PointerType.Touch || e.Pointer.Type == PointerType.Pen))
+            if (e.Pointer.Type == PointerType.Touch || e.Pointer.Type == PointerType.Pen)
```
Comment (kekekeks) [inline src/Avalonia.Base/Input/GestureRecognizers/ScrollGestureRecognizer.cs:69]: Are you sure that we want to intercept non-primary pointers for scrolling?

Comment (maxkatz6) [inline src/Avalonia.Base/Input/GestureRecognizers/ScrollGestureRecognizer.cs:69]:
```suggestion
            if (e.Pointer.IsPrimary && 
                (e.Pointer.Type == PointerType.Touch || e.Pointer.Type == PointerType.Pen))
```

Comment (maxkatz6) [reply src/Avalonia.Base/Input/GestureRecognizers/ScrollGestureRecognizer.cs]: We want, but after discussion it's better to revert it from this PR. I will come back to it in the future some day.

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0023008-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0023046-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0023223-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Review by kekekeks — APPROVED** [review]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0023231-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Pull request merged**